frontends/web/src/routes/account/send/components/dialogs/confirm-wait-dialog.tsx

```diff
@@ -0,0 +1,169 @@
+import { useTranslation } from 'react-i18next';
+import { WaitDialog } from '../../../../../components/wait-dialog/wait-dialog';
+import { TSignProgress } from '../../types';
+import { CoinCode, ConversionUnit, FeeTargetCode, Fiat, IAmount } from '../../../../../api/account';
+import { Amount } from '../../../../../components/amount/amount';
+import { customFeeUnit } from '../../../utils';
+import style from '../../send.module.css';
+
+type TransactionStatus = {
+  isConfirming: boolean;
+  signConfirm: boolean;
+  signProgress?: TSignProgress;
+}
+
+type TransactionDetails = {
+  proposedAmount?: IAmount;
+  proposedFee?: IAmount;
+  proposedTotal?: IAmount;
+  feeTarget?: FeeTargetCode;
+  customFee: string;
+  recipientAddress: string;
+  fiatUnit: Fiat;
+}
+
+type TProps = {
+  paired?: boolean;
+  baseCurrencyUnit: ConversionUnit;
+  note: string;
+  hasSelectedUTXOs: boolean;
+  selectedUTXOs: string[];
+  coinCode: CoinCode;
+  transactionStatus: TransactionStatus;
+  transactionDetails: TransactionDetails;
+
+}
```

Comment:
wdyt if we don't pass objects if not necessary?

```js
type TProps = TransactionStatus & TransactionDetails & {
  // ...
}

Reply:
```diff
diff --git a/frontends/web/src/routes/account/send/components/dialogs/confirm-wait-dialog.tsx b/frontends/web/src/routes/account/send/components/dialogs/confirm-wait-dialog.tsx
index 4678bc733..177580d8c 100644
--- a/frontends/web/src/routes/account/send/components/dialogs/confirm-wait-dialog.tsx
+++ b/frontends/web/src/routes/account/send/components/dialogs/confirm-wait-dialog.tsx
@@ -22,15 +22,13 @@ type TransactionDetails = {
   fiatUnit: Fiat;
 }
 
-type TProps = {
+type TProps = TransactionStatus & TransactionDetails & {
   paired?: boolean;
   baseCurrencyUnit: ConversionUnit;
   note: string;
   hasSelectedUTXOs: boolean;
   selectedUTXOs: string[];
   coinCode: CoinCode;
-  transactionStatus: TransactionStatus;
-  transactionDetails: TransactionDetails;
 
 }
 export const ConfirmingWaitDialog = ({
@@ -40,20 +38,19 @@ export const ConfirmingWaitDialog = ({
   hasSelectedUTXOs,
   selectedUTXOs,
   coinCode,
-  transactionStatus,
-  transactionDetails
+  isConfirming,
+  signConfirm,
+  signProgress,
+  proposedFee,
+  proposedAmount,
+  proposedTotal,
+  customFee,
+  feeTarget,
+  recipientAddress,
+  fiatUnit
 }: TProps) => {
   const { t } = useTranslation();
-  const { isConfirming, signConfirm, signProgress } = transactionStatus;
-  const {
-    proposedFee,
-    proposedAmount,
-    proposedTotal,
-    customFee,
-    feeTarget,
-    recipientAddress,
-    fiatUnit
-  } = transactionDetails;
+
 
   if (!isConfirming) {
     return null;
diff --git a/frontends/web/src/routes/account/send/send.tsx b/frontends/web/src/routes/account/send/send.tsx
index adbb36e43..cc73afb44 100644
--- a/frontends/web/src/routes/account/send/send.tsx
+++ b/frontends/web/src/routes/account/send/send.tsx
@@ -677,8 +677,8 @@ class Send extends Component<Props, State> {
                 hasSelectedUTXOs={this.hasSelectedUTXOs()}
                 selectedUTXOs={Object.keys(this.selectedUTXOs)}
                 coinCode={account.coinCode}
-                transactionDetails={waitDialogTransactionDetails}
-                transactionStatus={waitDialogTransactionStatus}
+                {...waitDialogTransactionDetails}
+                {...waitDialogTransactionStatus}
               />
               <MessageWaitDialog isShown={isSent} messageType={'sent'} />
               <MessageWaitDialog isShown={isAborted} messageType={'abort'} />
```

Reply:
Sure, we can also do that! Just a matter of preferences really. I like passing a whole object because of the fact:
1. we don't have a long prop for that component
2. the prop from the parent's side of that component, and that component's are matching.
IMO, these 2 things improve readability. 

But I have no objection with yours either 😄

Reply:
I'm not sure yet, but keep it as is for now. Btw could you make a new PR with the first 2 commits + the guide commit if possible?

Reply:
Yes, ofc! 😄

Reply:
#2193  :)